testcases/kernel/syscalls/fsetxattr/fsetxattr02.c

```diff
@@ -0,0 +1,267 @@
+ * License as published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it would be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ *
+ * Further, this software is distributed without any warranty that it
+ * is free of the rightful claim of any third person regarding
+ * infringement or the like.  Any license provided herein, whether
+ * implied or otherwise, applies only to this software file.  Patent
+ * licenses, if any, provided herein do not apply to combinations of
+ * this program with other software, or any other product whatsoever.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA
+ * 02110-1301, USA.
+ */
+
+/*
+ * In the user.* namespace, only regular files and directories can
+ * have extended attributes. Otherwise fsetxattr(2) will return -1
+ * and set errno to EPERM.
+ *
+ * There are 7 test cases:
+ * 1. Set attribute to a regular file, fsetxattr(2) should succeed
+ * 2. Set attribute to a directory, fsetxattr(2) should succeed
+ * 3. Set attribute to a symlink which points to the regular file,
+ *    fsetxattr(2) should return -1 and set errno to EEXIST
+ * 4. Set attribute to a FIFO, fsetxattr(2) should return -1 and set
+ *    errno to EPERM
+ * 5. Set attribute to a char special file, fsetxattr(2) should
+ *    return -1 and set errno to EPERM
+ * 6. Set attribute to a block special file, fsetxattr(2) should
+ *    return -1 and set errno to EPERM
+ * 7. Set attribute to a UNIX domain socket, fsetxattr(2) should
+ *    return -1 and set errno to EPERM
+ */
+
+#include "config.h"
+#include <sys/types.h>
+#include <sys/stat.h>
+#include <sys/sysmacros.h>
+#include <sys/wait.h>
+#include <errno.h>
+#include <fcntl.h>
+#include <unistd.h>
+#include <signal.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <sys/socket.h>
+#include <sys/un.h>
+#ifdef HAVE_SYS_XATTR_H
+# include <sys/xattr.h>
+#endif
+#include "tst_test.h"
+
+#ifdef HAVE_SYS_XATTR_H
+#define XATTR_TEST_KEY "user.testkey"
+#define XATTR_TEST_VALUE "this is a test value"
+#define XATTR_TEST_VALUE_SIZE 20
+
+#define FILENAME "fsetxattr02testfile"
+#define DIRNAME  "fsetxattr02testdir"
+#define SYMLINK  "fsetxattr02symlink"
+#define FIFO     "fsetxattr02fifo"
+#define CHR      "fsetxattr02chr"
+#define BLK      "fsetxattr02blk"
+#define SOCK     "fsetxattr02sock"
+
+struct test_case {
+	enum test_type {
+		TP_FILE,
+		TP_DIRECTORY,
+		TP_SYMLINK,
+		TP_FIFO,
+		TP_CHARDEV,
+		TP_BLOCKDEV,
+		TP_SOCKET
+	} type;
+	char *fname;
+	int filed;
+	int fflags;
+	char *key;
+	char *value;
+	size_t size;
+	int flags;
+	int exp_err;
+};
+static struct test_case tc[] = {
+	{			/* case 00, set attr to reg */
+	 .type = TP_FILE,
+	 .fname = FILENAME,
+	 .filed = -1,
```

Comment:
I guess that there is no point in setting the file descriptor to -1 here, since we check for > 0  in the test cleanup we can as well have it auto-initialized to 0.

Also fd is probably a bit more common name for a variable that represents file descriptor, but that is very minor...

Reply:
Agreed. Since I'll change this I'll rename the variable as well.